pandas/tseries/tests/test_resample.py

```diff
@@ -361,6 +361,29 @@ def test_fillna(self):
         with self.assertRaises(ValueError):
             r.fillna(0)
 
+    def test_fill_value(self):
```

Comment:
@jreback I'm a bit stuck on PeriodIndex. With my current commits you can add a fill_value for PeriodIndexResampler._upsample with kind='timestamp' but it doesn't work for kind='period'. I think I have to build that out.  Otherwise, this won't work for PeriodIndex.